File: 2021/05/16/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func minCameraCover(root *TreeNode) int {
	ans := 0
	if dfs(root, &ans) == 2 {
		ans++
	}
	return ans
}

func dfs(root *TreeNode, ans *int) int {
	if root == nil {
		return 1
	}
	l, r := dfs(root.Left, ans), dfs(root.Right, ans)
	if l == 2 || r == 2 {
		*ans++
		return 3
	} else if l == 3 || r == 3 {
		return 1
	} else {
		return 2
	}
}

File: 2021/05/16/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minCameraCover(self, root: TreeNode) -> int:
        self.ans = 0
        def dfs(root: TreeNode):
            if not root:
                return 1
            l = dfs(root.left)
            r = dfs(root.right)
            if l == 2 or r == 2:
                self.ans += 1
                return 3
            elif l == 3 or r == 3:
                return 1
            else:
                return 2
        if dfs(root) == 2:
            self.ans += 1
        return self.ans
